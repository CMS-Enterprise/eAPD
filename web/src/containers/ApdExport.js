import { Button } from '@cmsgov/design-system-core';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import Waypoint from './ConnectedWaypoint';
import { printApd } from '../actions/print';
import { Section } from '../components/Section';
import { FileDownload } from '../components/Icons';

const Bold = ({ children }) => (
  <span className="ds-u-font-weight--bold">{children}</span>
);
Bold.propTypes = { children: PropTypes.node.isRequired };

const Italic = ({ children }) => (
  <span className="ds-u-font-style--italic">{children}</span>
);
Italic.propTypes = { children: PropTypes.node.isRequired };

const ExportAndSubmit = ({ printApd: print }) => (
  <Waypoint id="export-and-submit">
    <Section isNumbered id="export-and-submit" resource="exportAndSubmit">
      <p>
        To submit your APD for analyst review, export the document as a PDF,
        then send it to CMS just like you’ve done before.
      </p>
      <ol>
        <li>
          Select <Bold>Export</Bold> to open up your browser’s print options.
        </li>
        <li>
          In the print window, select <Bold>Save as PDF</Bold>. In Chrome, this
          option is listed under <Italic>Destination</Italic>. In other
          browsers, this option is part of a dropdown menu labeled{' '}
          <Italic>PDF</Italic>.
        </li>

        <li>
          Select <Bold>Save</Bold>. This will download a PDF to your computer.
        </li>

        <li>
          If you decide to make changes before you submit, follow those three
          steps again to download a new PDF.
        </li>

        <li>
          Email the completed APD to{' '}
          <Bold>
            <a href="mailto:CMS-EAPD@cms.hhs.gov">CMS-EAPD@cms.hhs.gov</a>
          </Bold>
          .
        </li>
      </ol>
      <Button
        size="big"
        variation="primary"
        className="ds-u-margin-top--2"
        onClick={print}
      >
        Export
        <span className="ds-u-margin-left--2">
          <FileDownload />
        </span>
      </Button>
    </Section>
  </Waypoint>
);

ExportAndSubmit.propTypes = {
  printApd: PropTypes.func.isRequired
};

const mapDispatchToProps = { printApd };

export default connect(
  null,
  mapDispatchToProps
)(ExportAndSubmit);

export { ExportAndSubmit as plain, mapDispatchToProps };
