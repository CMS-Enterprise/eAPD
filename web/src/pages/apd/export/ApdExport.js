import PropTypes from 'prop-types';
import { Button } from '@cmsgov/design-system';
import React from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import { Redirect, useParams } from 'react-router-dom';

import { toggleAdminCheck } from '../../../redux/actions/app/apd';

import Icon, { faClipboardCheck } from '../../../components/Icons';

import { Section, Subsection } from '../../../components/Section';
import Waypoint from '../../../components/ConnectedWaypoint';
import AlertMissingFFY from '../../../components/AlertMissingFFY';
import { selectApdYears } from '../../../redux/selectors/apd.selectors';

import { useFlags } from 'launchdarkly-react-client-sdk';

const ExportAndSubmit = ({
  push: pushRoute,
  toggleAdminCheck: toggleAdmin,
  years
}) => {
  // inside the component code
  // Temporary feature flag
  const { adminCheckFlag } = useFlags();

  const paramApdId = useParams().apdId;

  if (!paramApdId) {
    return <Redirect to="/apd" />;
  }

  const handleAdminCheck = () => {
    toggleAdmin(true);
  };

  return (
    <React.Fragment>
      <Waypoint /> {/* Waypoint w/o id indicates top of page */}
      <AlertMissingFFY />
      <Section resource="exportAndSubmit">
        {adminCheckFlag === true && (
          <Subsection resource="adminCheck">
            <p>
              Choose Run Administrative Check to see a list of required fields
              which are missing content. Providing content will help ensure that
              your APD is administratively complete and ready for submission to
              CMS. Missing content from these sections could delay a decision by
              CMS and result in additional review cycles for this APD.
            </p>
            <Button
              onClick={handleAdminCheck}
              size="big"
              variation="primary"
              className="ds-u-margin-bottom--2"
            >
              Run Administrative Check
              <Icon className="ds-u-padding-left--1" icon={faClipboardCheck} />
            </Button>
          </Subsection>
        )}
        <Subsection resource="reviewAndDownload">
          <p>
            On the next page, you will be able to review and download a copy of
            your APD as the first step in submitting a completed APD to CMS.
          </p>
          <Button
            size="big"
            variation="primary"
            className="ds-u-margin-top--2"
            onClick={() =>
              years.length > 0 && pushRoute(`/print/${paramApdId}`)
            }
            disabled={years.length === 0}
          >
            Continue to Review
          </Button>
        </Subsection>
      </Section>
    </React.Fragment>
  );
};

ExportAndSubmit.propTypes = {
  push: PropTypes.func.isRequired,
  years: PropTypes.arrayOf(PropTypes.string).isRequired,
  toggleAdminCheck: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  years: selectApdYears(state)
});

const mapDispatchToProps = {
  push,
  toggleAdminCheck
};

export default connect(mapStateToProps, mapDispatchToProps)(ExportAndSubmit);

export { ExportAndSubmit as plain, mapDispatchToProps };
