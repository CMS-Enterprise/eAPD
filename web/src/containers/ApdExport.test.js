import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';
import { push } from 'connected-react-router';

import { plain as ExportAndSubmit, mapDispatchToProps } from './ApdExport';

describe('apd export component', () => {
  test('renders correctly', () => {
    expect(shallow(<ExportAndSubmit printApd={() => {}} />)).toMatchSnapshot();
  });

  test('routes to print preview', () => {
    const fakePush = sinon.spy();
    const component = shallow(<ExportAndSubmit push={fakePush} />);
    component.find('Button').simulate('click');

    expect(fakePush.calledOnce).toEqual(true);
  });

  test('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({ push });
  });
});
