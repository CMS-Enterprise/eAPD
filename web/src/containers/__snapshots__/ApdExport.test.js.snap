// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd export component renders correctly 1`] = `
<Connect(ConnectedWaypoint)
  id="export-and-submit"
>
  <Section
    id="export-and-submit"
    isNumbered={true}
    resource="exportAndSubmit"
  >
    <h3
      className="ds-h3"
    >
      Review and Download
    </h3>
    <p>
      On the next page, you\`'\`ll be able to review and download a copy of your APD as the first step in submitting a completed APD to CMS.
    </p>
    <Button
      className="ds-u-margin-top--2"
      onClick={[Function]}
      size="big"
      type="button"
      variation="primary"
    >
      Continue to Review
    </Button>
  </Section>
</Connect(ConnectedWaypoint)>
`;
